### DISCUSSION TITLE
Translations

### DISCUSSION THREAD

**Discussion by ukraine-sdg** [open-sdg/open-sdg#1504] [Q&A]
Could you, please, help with 2 issues:

(1) How can I write on the frontpage in Ukrainian on the Ukrainian version of the site this text.
"Ukrainian data for Sustainable Development Goal indicators
Testing stuff that looks like stuff"?
I can not find a sections on the weblate with this text

(2) How can I edit translations? I have suggested changes for translation of some text on the front page. Should I do some additional steps to have the edited translation of the site?

(3) how can I translate the content of the "about" 

(4) how can I edit a FAQ section and translate it

(5) how to edit text "lorem ipsum ..." on the front page and add a link from this text

(6) how to add logos on the front page

Thank you

**Comment by brockfanning** [comment]
@ukraine-sdg 

> (1) How can I write on the frontpage in Ukrainian on the Ukrainian version of the site this text. "Ukrainian data for Sustainable Development Goal indicators Testing stuff that looks like stuff"? I can not find a sections on the weblate with this text

These items are controlled in your [site configuration](https://ukraine-sdg.github.io/site/config/) under "Frontpage goals grid". In order to translate those, you would need to change them to "translation keys". One way to do this would be the following:

First, create a translation file in your data repository, called `translations/en/custom.yml`. The contents could be:

         frontpage_goals_grid_title: My English translation of the title
         frontpage_goals_grid_description: My English translation of the description

Next, create a similar file for the Ukraine version, called `translations/uk/custom.yml`, with appropriate translations.

Finally, update your site configuration:

* Set the frontpage goals grid "Title" to: `custom.frontpage_goals_grid_title`
* Set the frontpage goals grid "Description" to: `custom.frontpage_goals_grid_description`

> (2) How can I edit translations? I have suggested changes for translation of some text on the front page. Should I do some additional steps to have the edited translation of the site?

In general, if the text is specific to your implementation, then you would need to add the translations to your data repository, as described above. However, if you notice that any of the translations in the [sdg-translations project](https://hosted.weblate.org/projects/sdg-translations/) are wrong, then we certainly appreciate any corrections you can make there.

> (3) how can I translate the content of the "about"

For translating page content, you would edit that file directly in your site repository. For example, to edit the Ukrainian "About" page, you would edit this file: https://github.com/ukraine-sdg/site/blob/develop/_pages/uk/about.md

> (4) how can I edit a FAQ section and translate it

Same as described above, you would edit the files in your site repository. To your question, you would edit the [English FAQ file](https://github.com/ukraine-sdg/site/blob/develop/_pages/faq.md) and the [Ukrainian FAQ file](https://github.com/ukraine-sdg/site/blob/develop/_pages/uk/faq.md).

> (5) how to edit text "lorem ipsum ..." on the front page and add a link from this text

This text is controlled in the site configuration under "Frontpage cards". Just as with the "Frontpage goals grid" described above, to translate that configuration you would need to use "translation keys" which correspond to entries in translation files, like the `custom.yml` files mentioned above.

> (6) how to add logos on the front page

I need to clarify the question: Are you talking about the main logo at the top of each page? Or are you talking about the logo to the right of the "17 goals to transform our world"? Or are you asking about adding logos inside the "frontpage cards" mentioned above?

**Reply by ukraine-sdg** [reply]
Thank you, Brock!
I wasasking about the logo to the right of the "17 goals to transform our world"?

**Comment by brockfanning** [comment]
@ukraine-sdg That image is in the open-sdg repository in the "assets/img" folder and it is called "SDG Wheel_Transparent-01.png". ([Here is a direct link to the file in the open-sdg repository](https://github.com/open-sdg/open-sdg/blob/1.7.0-dev/assets/img/SDG%20Wheel_Transparent-01.png).

Like anything in that assets folder, you can "override" it by placing a file with the same name into your site repository. So, to override that file, you would create your own version -- also called "SDG Wheel_Transparent-01.png" -- and upload it to your site repository in the [assets/img](https://github.com/ukraine-sdg/site/tree/develop/assets/img) folder.

Then your version would override the version in the open-sdg repository.

EDIT: Fixed filename

**Reply by ukraine-sdg** [reply]
Thank you, Brock!


And I also can not find a way to correct translation for:
"The Sustainable Development Goals (SDGs) are a universal call to action to end poverty, protect the planet and improve the lives and prospects of everyone, everywhere. The 17 Goals were adopted by all UN Member States in 2015, as part of the 2030 Agenda for Sustainable Development."

**Comment by ukraine-sdg** [comment]
thank you!

And I also can not find a way to correct translation for:
"The Sustainable Development Goals (SDGs) are a universal call to action to end poverty, protect the planet and improve the lives and prospects of everyone, everywhere. The 17 Goals were adopted by all UN Member States in 2015, as part of the 2030 Agenda for Sustainable Development."

**Comment by brockfanning** [comment]
@ukraine-sdg That content is coming from the SDG Translations project, via the translation key "frontpage.intro_body". If the Ukrainian translation is wrong, we would appreciate if you could correct it here: https://hosted.weblate.org/translate/sdg-translations/frontpage/uk/?checksum=dfd6abfb7b00556f

On the other hand, if the translation is "correct" but you would still like to customize it, you could create your own translation key, and add it to your own translation files, like described above. Similar to above, the steps would be:

First, create a translation file in your data repository, called `translations/en/custom.yml`. The contents could be:

         frontpage_intro_banner_body: My English translation of the intro body

Next, create a similar file for the Ukraine version, called `translations/uk/custom.yml`, with appropriate translations.

Finally, update your site configuration:

* Set the frontpage introduction banner "Description" to: `custom.frontpage_intro_banner_body`

**Reply by ukraine-sdg** [reply]
Dear Brock, I tried to replace the logo on the right coner (the wheel). I relaced it in the site repository, but it was not replaced on the site.
Could you, please, take a look?

**Comment by brockfanning** [comment]
@ukraine-sdg I'm sorry - I got the name of the file wrong. It should be: `SDG Wheel_Transparent-01.png`

**Reply by ukraine-sdg** [reply]
thank you, Brock!
it works!
